File: geqiangbao_20201117_CommandPattern_01/geqiangbao_20201117_CommandPattern_01/Factory/HW/HWFactory.m
//
//  HWFactory.m
//  geqiangbao_20201117_CommandPattern_01
//
//  Created by xiaobao on 2020/11/19.
//

#import "HWFactory.h"

@implementation HWFactory

- (id<ComputerProtocol>)getComputer {
    return [[HWComputer alloc]init];
}

@end

File: 22.工厂模式(3种原理案例)/geqiangbao_20201207_factory_swift/geqiangbao_20201207_factory_swift/工厂模式/3.抽象工厂模式/HW/HWFactory.swift
//
//  HWFactory.swift
//  geqiangbao_20201207_factory_swift
//
//  Created by xiaobao on 2020/12/7.
//

import UIKit

/// 华为工厂
class HWFactory: NSObject,FactoryProtocol {

    func getComputer() -> ComputerProtocol {
        return HWComputer()
    }
    
    func getMobilePhone() -> MobilePhoneProtocol {
        return HWMobilePhone()
    }
    
}
